### PR TITLE
Network setup update

### PR DESCRIPTION
**Pull request by unixpi**
move networking tips under one page; refactor FAQ; add logging info

### CODE REVIEW

File: docs/the_nimbus_book/src/faq.md [hunk 1]
```diff
@@ -59,6 +46,26 @@ To add an additional validator, just follow [the same steps](./keys.md) as you d
...
+
+See [here](./networking.md).
+
+### How do I fix the discovered new external address warning log?
+ 
+```
+WRN 2021-03-15 02:23:37.569+00:00 Discovered new external address but ENR auto update is off topics="discv5"...
+```
+
+It's possible that your ISP has changed your IP address without you knowing.
```
Comment (stefantalpalaru) [inline docs/the_nimbus_book/src/faq.md]:
```suggestion
It's possible that your ISP has changed your dynamic IP address without you knowing.
```

File: docs/the_nimbus_book/src/faq.md [hunk 2]
```diff
@@ -59,6 +46,26 @@ To add an additional validator, just follow [the same steps](./keys.md) as you d
...
+
+The first thing to do it to try relaunching the beacon node with with `--enr-auto-update:true` (pass it as an option in the command line).
```
Comment (stefantalpalaru) [inline docs/the_nimbus_book/src/faq.md]:
```suggestion
The first thing to do it to try relaunching the beacon node with `--enr-auto-update` (pass it as an option in the command line).
```

File: docs/the_nimbus_book/src/networking.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# Network setup
+
+
+A collection of tips and tricks to help improve your network connectivity.
+
+## Monitor your Peer count
+
+If your Peer count is low (less than `20`) and/or you repeatedly see the following warning:
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion
If your Peer count is low (less than `15`) and/or you repeatedly see the following warning:
```

File: docs/the_nimbus_book/src/networking.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+If you're running on a home network and want to ensure you are able to receive incoming connections you may need to set up port forwarding (though some routers automagically set this up for you).
+
+
+> **Note:** If you are running your node on a virtual public cloud (VPC) instance, you can safely ignore this section.
+
+While the specific steps required vary based on your router, they can be summarised as follows:
+
+
+1. Determine your [public IP address](./health.md#public-ip-address)
+2. Determine your [private IP address](./health.html#private-ip-address)
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion
2. Determine your [private IP address](./health.md#private-ip-address)
```

Comment (arnetheduck) [reply docs/the_nimbus_book/src/networking.md]: actually, these links should probably live in networking now?

File: docs/the_nimbus_book/src/networking.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+3. Browse to the management website for your home router (typically [http://192.168.1.1)](http://192.168.1.1)
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion
3. Browse to the management website for your home router (typically [http://192.168.1.1](http://192.168.1.1))
```

File: docs/the_nimbus_book/src/networking.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+If you're still experiencing a low peer count, you may be behind a firewall. Try restarting your client and passing `--nat:extip:$EXT_IP_ADDRESS` as an option to the client,  where `$EXT_IP_ADDRESS` is your real IP. For example, if your real IP address is `35.124.65.104`, you'd run:
+
+```
+./run-prater-beacon-node.sh --nat:extip:35.124.65.104
+```
+
+## ENR auto update = true
+
+Additionally you can turn on `--enr-auto-update` to allow discovery to keep your external IP address up to date based on information received from peers.
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion

The `--enr-auto-update` feature keeps your external IP address up to date based on information received from other peers on the network. This option is useful with ISPs that assign IP addresses dynamically.
```

File: docs/the_nimbus_book/src/networking.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+ipconfig | findstr /i "IPv4 Address"
+```
+
+**macOS:**
+
+```
+ifconfig | grep "inet " | grep -v 127.0.0.1
+```
+
+### Check open ports on your connection
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]: newline

File: docs/the_nimbus_book/src/networking.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+Use [this tool](https://www.yougetsignal.com/tools/open-ports/) to check your external (public) IP address and detect open ports on your connection (Nimbus TCP and UDP ports are both set to `9000` by default).
+
+
+## Pass the extip option
+
+If you're still experiencing a low peer count, you may be behind a firewall. Try restarting your client and passing `--nat:extip:$EXT_IP_ADDRESS` as an option to the client,  where `$EXT_IP_ADDRESS` is your real IP. For example, if your real IP address is `35.124.65.104`, you'd run:
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion
If you have a static public IP address, use the `--nat:extip:$EXT_IP_ADDRESS` option to pass it to the client,  where `$EXT_IP_ADDRESS` is your public IP. For example, if your public IP address is `1.2.3.4`, you'd run:
```

File: docs/the_nimbus_book/src/networking.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
+# Network setup
+
+
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md:4]:
```suggestion

Nimbus will automatically connect to peers based on the health and quality of peers that it's already connected to. Depending on the network and the number of validators attached to the the node, Nimbus may need anywhere from 10 to 60 peers connected to operate well.

In addition to making outgoing connections, the beacon node node works best when others can connect to the node - this speeds up the process of finding good peers.

To allow incoming connections, the peer must be reachable via a public IP address.

By default, Nimbus uses UPnP to set up port forwarding and detect your external IP address. If you do not have UPnP enabled, you may need to pass additional command-line options to the node, as detailed below.
```

File: docs/the_nimbus_book/src/networking.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+A collection of tips and tricks to help improve your network connectivity.
+
+## Monitor your Peer count
+
+If your Peer count is low (less than `20`) and/or you repeatedly see the following warning:
+```
+WRN 2021-05-08 12:59:26.669+00:00 Peer count low, no new peers discovered    topics="networking" tid=1914 file=eth2_network.nim:963 discovered_nodes=9 new_peers=0 current_peers=1 wanted_peers=160
+```
+
+It probably means that your computer is not reachable from the outside. This means you won't be able to accept any incoming peer connections.
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]:
```suggestion
It means that Nimbus was unable to find a sufficient number of peers to guarantee stable operation, and you may miss attestations and blocks as a result. 

Most commonly, this happens when your computer is not reachable from the outside and therefore won't be able to accept any incoming peer connections.
```

File: docs/the_nimbus_book/src/networking.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+If you're on a home network, the fix here is to set up port forwarding.
+
+## Check for incoming connections
+
+To check if you have incoming connections set, run:
+
+```
+curl -s http://localhost:8008/metrics | grep libp2p_open_streams
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md:32]: need an example here of what it looks like when there are no incoming connections (in = 0..)

File: docs/the_nimbus_book/src/faq.md [hunk 12]
```diff
@@ -4,7 +4,7 @@
 
 ### How do I check which version of Nimbus I'm currently running?
 
-If you'v enabled RPC, the version is available via
+If you've enabled RPC, the version is available via
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/faq.md:7]: This should be updated to use REST instead

File: docs/the_nimbus_book/src/networking.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+./run-prater-beacon-node.sh --nat:extip:35.124.65.104
+```
+
+## ENR auto update = true
+
+Additionally you can turn on `--enr-auto-update` to allow discovery to keep your external IP address up to date based on information received from peers.
+
+In practice this means relaunching the beacon node with `--enr-auto-update:true` (pass it as an option in the command line).
+
+## Reading the logs
```
Comment (arnetheduck) [inline docs/the_nimbus_book/src/networking.md]: `No peers for topic, skipping publish` is printed when we lack quality peers to publish attestations to - this is the most important indication that the node is having trouble keeping up

Comment (arnetheduck) [reply docs/the_nimbus_book/src/networking.md]: Basically, the "low peers warning" means that you *may* lose attestations - `skipping publish` instead means you lost one for sure

File: docs/the_nimbus_book/src/networking.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+
+```
+ifconfig | grep "inet " | grep -v 127.0.0.1
+```
+
+### Check open ports on your connection
+Use [this tool](https://www.yougetsignal.com/tools/open-ports/) to check your external (public) IP address and detect open ports on your connection (Nimbus TCP and UDP ports are both set to `9000` by default).
+
+
+## Pass the extip option
```
Comment (kdeme) [inline docs/the_nimbus_book/src/networking.md]: I think this should go before `### Check open ports on your connection`, and thus as part of that section?
If you are explaining how to set up port forwarding, figure out public IP address, etc. then it means you are behind NAT but without UPnP working and thus you need `--nat:extip:<ip addr>` and/or `--enr-auto-update`.

File: docs/the_nimbus_book/src/networking.md (hunk 8 above)
Comment (kdeme) [reply docs/the_nimbus_book/src/networking.md]: You can also still use it with a dynamic IP address. Just that you will probably also require `--enr-auto-update` in that case.

File: docs/the_nimbus_book/src/networking.md (hunk 11 above)
Comment (unixpi) [reply docs/the_nimbus_book/src/networking.md:32]: `libp2p_open_streams{type="LPChannel",dir="In"} 0` ?

Comment (arnetheduck) [reply docs/the_nimbus_book/src/networking.md:32]: the `ChronosStream` one

**Pull request merged**